### PR TITLE
implemented mnist training test, and unified mnist example/test utili…

### PR DESCRIPTION
**Pull request by baykaner**
…ties

### CODE REVIEW

**Review by MissingNO57 — DISMISSED** [review]
LGTM

File: libs/ml/include/ml/utilities/mnist_utilities.hpp [hunk 1]
```diff
@@ -18,13 +18,70 @@
...
+  ModelType model(model_config);
+  model.template Add<fetch::ml::layers::FullyConnected<TensorType>>(
+      784, 100, fetch::ml::details::ActivationType::RELU);
+  model.template Add<fetch::ml::layers::FullyConnected<TensorType>>(
+      100, 20, fetch::ml::details::ActivationType::RELU);
+  model.template Add<fetch::ml::layers::FullyConnected<TensorType>>(
+      20, 10, fetch::ml::details::ActivationType::RELU);
+
+  model.SetDataloader(std::move(data_loader_ptr));
+  model.Compile(OptimiserType::ADAM, LossType::MEAN_SQUARE_ERROR);
```
Comment (evsmithx) [inline libs/ml/include/ml/utilities/mnist_utilities.hpp]: Shouldn't it be cross entropy loss?

File: libs/ml/tests/integration/mnist/mnist.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  model_config.learning_rate_param.exponential_decay_rate = fetch::math::Type<DataType>("0.97");
+
+  // generate some random data for training and testing
+  auto data_label_pair = fetch::ml::utilities::generate_dummy_data<TypeParam>(N_DATA);
+
+  // build and compile and mnist classifier
+  auto model = fetch::ml::utilities::setup_mnist_model<TypeParam>(
+      model_config, data_label_pair.first, data_label_pair.second);
+
+  // training loop - early stopping will prevent long training time
```
Comment (evsmithx) [inline libs/ml/tests/integration/mnist/mnist.cpp]: Is early stopping applied here?

**Review by evsmithx — CHANGES_REQUESTED** [review]

File: libs/ml/include/ml/utilities/mnist_utilities.hpp (hunk 1 above)
Comment (baykaner) [reply libs/ml/include/ml/utilities/mnist_utilities.hpp]: oops!  yep

**Review by evsmithx — DISMISSED** [review]

**Review by evsmithx — APPROVED** [review]
